ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -939,6 +939,26 @@ mkSnocket state tr = Snocket { getLocalAddr
                              (TestAddress addr)
     accept FD { fdVar } = accept_
       where
+        readTBQueueUntil :: (a -> STM m Bool) -> TBQueue m a -> STM m a
+        readTBQueueUntil p queue = do
+          a <- readTBQueue queue
+          shouldReturn <- p a
+          if shouldReturn
+             then return a
+             else readTBQueueUntil p queue
+
+        isHalfOpened :: TestAddress addr
+                     -> ChannelWithInfo m (TestAddress addr)
+                     -> STM m Bool
+        isHalfOpened localAddress cwi = do
+          connMap <- readTVar (nsConnections state)
+          let connId = ConnectionId localAddress (cwiAddress cwi)
+
+          case Map.lookup (normaliseId connId) connMap of
+             Nothing                            -> return True
```

Comment:
Actually we shouldn't error here. We should return True because we might be accepting a connection from a peer which we still do not know and haven't made an outbound request to.

Reply:
ok, you can leave it as that.  I have some other changes on top of this branch which clean this part. Here's  the argument though:  the reason that we should not wait until unknown connection is that it's a simulation error to accept a connection which is unknown to the simulation.   In my branch I return `False` here, but I changed some other places which could lead to a situation where `connect` can delete the connection from the simulation state.  This could happen after a connection is written to the accept queue, hence we need to ignore it.